# Exploratory Data Analysis (EDA)

This notebook examines the smartphone addiction dataset to understand its structure, data quality, feature distributions, and missingness patterns — establishing the context needed for modelling decisions.

---

## Table of Contents

1. [Setup and Data Loading](#eda-setup-and-data-loading)
   - 1.1 [Load Datasets](#eda-load-datasets)
2. [Dataset Overview](#eda-dataset-overview)
   - 2.1 [Dataset Dimensions](#eda-dataset-dimensions)
   - 2.2 [Data Preview](#eda-data-preview)
   - 2.3 [Column Overview](#eda-column-overview)
   - 2.4 [Data Types and Missingness](#eda-data-types-and-missingness)
3. [Target Distribution](#eda-target-distribution)
4. [Categorical Features](#eda-categorical-features)
5. [Numerical Features](#eda-numerical-features)
6. [Missing Values: Train vs. Test](#eda-missing-values-train-vs-test)
7. [Summary](#eda-summary)

---

<a id="eda-setup-and-data-loading" name="eda-setup-and-data-loading"></a>

## 1. Setup and Data Loading

In [1]:
from pathlib import Path

import pandas as pd

<a id="eda-load-datasets" name="eda-load-datasets"></a>

### 1.1 Load Datasets

In [2]:
PROJECT_ROOT = Path.cwd().parent
DATA_DIR = PROJECT_ROOT / "data"

train = pd.read_csv(DATA_DIR / "train.csv")
test = pd.read_csv(DATA_DIR / "test.csv")
sample_submission = pd.read_csv(DATA_DIR / "sample_submission.csv")

print("Project root:", PROJECT_ROOT)
print("Data dir:", DATA_DIR)

Project root: /Users/c.c./Developer/Competitions/predicting-smartphone-addiction
Data dir: /Users/c.c./Developer/Competitions/predicting-smartphone-addiction/data


<a id="eda-dataset-overview" name="eda-dataset-overview"></a>

## 2. Dataset Overview

<a id="eda-dataset-dimensions" name="eda-dataset-dimensions"></a>

### 2.1 Dataset Dimensions

In [3]:
print("Train shape:", train.shape)
print("Test shape:", test.shape)
print("Submission shape:", sample_submission.shape)

Train shape: (691369, 14)
Test shape: (296302, 13)
Submission shape: (296302, 2)


<a id="eda-data-preview" name="eda-data-preview"></a>

### 2.2 Data Preview

In [4]:
train.head(10)

,id,age,daily_screen_time_hours,social_media_hours,gaming_hours,work_study_hours,sleep_hours,notifications_per_day,app_opens_per_day,weekend_screen_time,gender,stress_level,academic_work_impact,addicted_label
0,0,24.0,NaN,1.83,1.59,2.11,7.46,122.0,38.0,8.63,Male,Medium,No,1
1,1,19.0,5.97,1.08,NaN,3.03,8.22,76.0,19.0,NaN,Female,Medium,No,0
2,2,18.0,5.09,NaN,NaN,NaN,6.25,134.0,60.0,7.47,Female,Low,Yes,0
3,3,21.0,6.42,1.26,1.42,3.36,8.85,112.0,94.0,8.66,Other,Low,NaN,1
4,4,26.0,11.20,1.87,2.81,1.95,5.25,NaN,NaN,13.39,Female,Medium,No,1
5,5,28.0,7.37,1.78,1.56,3.38,8.33,210.0,40.0,7.85,Male,High,No,0
6,6,18.0,4.24,1.61,1.14,0.91,5.09,220.0,68.0,7.46,Other,Low,Yes,0
7,7,25.0,10.79,3.92,2.48,3.61,7.20,195.0,NaN,13.45,Other,High,No,1
8,8,26.0,8.42,2.57,3.41,1.90,6.64,99.0,27.0,8.35,Other,Medium,Yes,0
9,9,20.0,8.74,1.40,2.43,2.11,8.57,NaN,109.0,10.68,Other,High,NaN,1


In [5]:
display(test.head())

,id,age,daily_screen_time_hours,social_media_hours,gaming_hours,work_study_hours,sleep_hours,notifications_per_day,app_opens_per_day,weekend_screen_time,gender,stress_level,academic_work_impact
0,691369,30.0,9.34,NaN,0.77,4.09,7.15,153.0,16.0,NaN,Other,Medium,Yes
1,691370,NaN,NaN,1.91,NaN,1.20,8.67,239.0,148.0,10.68,Female,High,No
2,691371,26.0,8.48,3.64,NaN,3.39,7.47,106.0,123.0,NaN,Male,NaN,No
3,691372,20.0,8.37,2.99,1.69,2.53,5.45,178.0,55.0,9.88,Male,High,Yes
4,691373,25.0,8.25,4.02,1.21,2.32,8.46,63.0,86.0,10.72,Male,Low,No


In [6]:
display(sample_submission.head())

,id,addicted_label
0,691369,0.709424
1,691370,0.709424
2,691371,0.709424
3,691372,0.709424
4,691373,0.709424


<a id="eda-column-overview" name="eda-column-overview"></a>

### 2.3 Column Overview

In [7]:
print(train.columns.tolist())
print()
print(*train.columns, sep="\n")
# for col in train.columns:
#     print(col)

['id', 'age', 'daily_screen_time_hours', 'social_media_hours', 'gaming_hours', 'work_study_hours', 'sleep_hours', 'notifications_per_day', 'app_opens_per_day', 'weekend_screen_time', 'gender', 'stress_level', 'academic_work_impact', 'addicted_label']

id
age
daily_screen_time_hours
social_media_hours
gaming_hours
work_study_hours
sleep_hours
notifications_per_day
app_opens_per_day
weekend_screen_time
gender
stress_level
academic_work_impact
addicted_label


In [8]:
print("Train-only columns:",set(train.columns) - set(test.columns))

Train-only columns: {'addicted_label'}


<a id="eda-data-types-and-missingness" name="eda-data-types-and-missingness"></a>

### 2.4 Data Types and Missingness

In [9]:
train.info()  # Numeric columns with missing values may be inferred as float.

<class 'pandas.DataFrame'>
RangeIndex: 691369 entries, 0 to 691368
Data columns (total 14 columns):
 #   Column                   Non-Null Count   Dtype  
---  ------                   --------------   -----  
 0   id                       691369 non-null  int64  
 1   age                      662440 non-null  float64
 2   daily_screen_time_hours  595515 non-null  float64
 3   social_media_hours       557374 non-null  float64
 4   gaming_hours             564548 non-null  float64
 5   work_study_hours         639851 non-null  float64
 6   sleep_hours              646889 non-null  float64
 7   notifications_per_day    623785 non-null  float64
 8   app_opens_per_day        610659 non-null  float64
 9   weekend_screen_time      579306 non-null  float64
 10  gender                   662335 non-null  str    
 11  stress_level             636221 non-null  str    
 12  academic_work_impact     647145 non-null  str    
 13  addicted_label           691369 non-null  int64  
dtypes: float64(9), 

In [10]:
summary = pd.DataFrame({
    "dtype": train.dtypes,
    "missing": train.isna().sum(),
    "missing_pct": (train.isna().mean() * 100).round(2),
    "nunique": train.nunique()  # unique values,ignore NaN
})

display(summary)

,dtype,missing,missing_pct,nunique
id,int64,0,0.00,691369
age,float64,28929,4.18,18
daily_screen_time_hours,float64,95854,13.86,1389
social_media_hours,float64,133995,19.38,721
gaming_hours,float64,126821,18.34,401
work_study_hours,float64,51518,7.45,600
sleep_hours,float64,44480,6.43,451
notifications_per_day,float64,67584,9.78,231
app_opens_per_day,float64,80710,11.67,166
weekend_screen_time,float64,112063,16.21,1437


<a id="eda-target-distribution" name="eda-target-distribution"></a>

## 3. Target Distribution

In [11]:
ID_COL = "id"

target_counts = train["addicted_label"].value_counts()
target_ratio = train["addicted_label"].value_counts(normalize=True)

print("Target counts:",target_counts,"\n")
print("Target ratio:",target_ratio.round(3))

Target counts: addicted_label
1    490474
0    200895
Name: count, dtype: int64 

Target ratio: addicted_label
1    0.709
0    0.291
Name: proportion, dtype: float64


<a id="eda-categorical-features" name="eda-categorical-features"></a>

## 4. Categorical Features

In [12]:
cat_cols = [
    "gender",
    "stress_level",
    "academic_work_impact",
]

for col in cat_cols:
    print(f"\n===== {col} =====")
    print(train[col].value_counts(dropna=False))


===== gender =====
gender
Male      223662
Female    221595
Other     217078
NaN        29034
Name: count, dtype: int64

===== stress_level =====
stress_level
High      220873
Low       207783
Medium    207565
NaN        55148
Name: count, dtype: int64

===== academic_work_impact =====
academic_work_impact
Yes    330566
No     316579
NaN     44224
Name: count, dtype: int64


<a id="eda-numerical-features" name="eda-numerical-features"></a>

## 5. Numerical Features

In [13]:
num_cols = [
    col for col in train.columns
    if col not in cat_cols + ["addicted_label", "id"]
]

print(train[num_cols].describe().T)

# num_cols = train.select_dtypes(include="number").columns.difference(["addicted_label", "id"])

# display(train[num_cols].describe().T)

                            count        mean        std    min    25%  \
age                      662440.0   26.615408   5.153162  18.00  22.00   
daily_screen_time_hours  595515.0    7.640865   2.721446   0.50   5.48   
social_media_hours       557374.0    2.471038   1.316137   0.00   1.45   
gaming_hours             564548.0    1.459265   0.934552   0.00   0.70   
work_study_hours         639851.0    2.366971   1.258797   0.00   1.36   
sleep_hours              646889.0    6.804334   1.234512   4.50   5.78   
notifications_per_day    623785.0  145.894900  65.917556  20.00  93.00   
app_opens_per_day        610659.0  102.636781  48.093970  15.00  64.00   
weekend_screen_time      579306.0    9.479866   2.856006   0.51   7.28   

                            50%     75%     max  
age                       27.00   31.00   35.00  
daily_screen_time_hours    7.77    9.84   15.00  
social_media_hours         2.31    3.37    8.00  
gaming_hours               1.33    2.09    4.00  
work_stud

<a id="eda-missing-values-train-vs-test" name="eda-missing-values-train-vs-test"></a>

## 6. Missing Values: Train vs. Test

In [14]:
missing_compare = pd.DataFrame({
    "train_missing_pct": (train.drop(columns="addicted_label").isna().mean() * 100).round(3),
    "test_missing_pct": (test.isna().mean() * 100).round(3),
})

missing_compare["difference"] = (
    missing_compare["test_missing_pct"]
    - missing_compare["train_missing_pct"]
).round(3)

display(
    missing_compare.sort_values(
        "train_missing_pct",
        ascending=False
    )
)

,train_missing_pct,test_missing_pct,difference
social_media_hours,19.381,15.996,-3.385
gaming_hours,18.343,20.054,1.711
weekend_screen_time,16.209,17.110,0.901
daily_screen_time_hours,13.864,11.066,-2.798
app_opens_per_day,11.674,8.675,-2.999
notifications_per_day,9.775,11.549,1.774
stress_level,7.977,6.624,-1.353
work_study_hours,7.452,9.375,1.923
sleep_hours,6.434,7.578,1.144
academic_work_impact,6.397,8.681,2.284


<a id="eda-summary" name="eda-summary"></a>

## 7. Summary

| Area | Main finding |
|---|---|
| Dataset | 691,369 training rows, 296,302 test rows, and 12 predictive features |
| Target | 70.9% positive vs. 29.1% negative |
| Missingness | 4.2%–19.4% across all predictive features; the largest train–test gap is 3.4 percentage points |

### Modelling implications

- The target imbalance makes stratified validation important, while the moderate train–test missingness differences should be monitored during evaluation.
- With 9 numerical features, 3 categorical features, and widespread missing values, CatBoost is a natural baseline because it can preserve numerical `NaN`s and handle categorical variables directly.

### Next direction

Build a simple CatBoost baseline first, using explicit missing-value labels for categorical columns and leaving numerical missing values to the model.